### DISCUSSION TITLE
Iterative Quantum Phase Estimation bug

### DISCUSSION THREAD

**Discussion by debaarjun13** [Qiskit/.github#38] [General Feedback]
I believe in the new module for the Iterative Quantum Phase Estimation there is a problem where there is a reordering of the qubits of the unitary operator when the circuit is being formed. The module runs fine for a 2 qubit circuit but creates problems when more than 2 qubits are involved .
Is it mainly due to the transpilation done during run time?
I tackled the issue by coding the unitary operator in a reversed manner and then flipping the ordering of the qubits in the source code (it is funny how the reverse ordering does not cause any problems during circuit composing)

**Comment by 1ucian0** [comment]
The `IterativePhaseEstimation` class currently lives in [qiskit-algorithms](https://github.com/qiskit-community/qiskit-algorithms). Consider submitting it to https://github.com/qiskit-community/qiskit-algorithms/issues
